### PR TITLE
Reset nursery_index in finalizable processor if we remove objects from candidates

### PR DESCRIPTION
**Pull request by qinsoon**

### CODE REVIEW

File: src/util/finalizable_processor.rs [hunk 1]
```diff
@@ -72,6 +72,7 @@ impl<F: Finalizable> FinalizableProcessor<F> {
         // Keep the finalizable objects alive.
         self.forward_finalizable(e, nursery);
 
+        // Set nursery_inex to the end of the candidates (the candidates before the index are scanned)
```
Comment (wks) [inline src/util/finalizable_processor.rs:75]:
```suggestion
        // Set nursery_index to the end of the candidates (the candidates before the index are scanned)
```

**Review by wks — CHANGES_REQUESTED** [review]
Just a minor spelling mistake.

File: src/util/finalizable_processor.rs (hunk 1 above)
Comment (qinsoon) [reply src/util/finalizable_processor.rs]: Done

**Review by wks — APPROVED** [review]
LGTM

**Pull request merged**